documentation/quarkus/guide-beanmapping-quarkus.asciidoc

```diff
@@ -0,0 +1,224 @@
+:toc: macro
+toc::[]
+
+= Bean mapping with Quarkus
+
+This guide will show bean mapping in particular for a Quarkus application. To get a quick introduction to Bean Mapping you can read this https://github.com/devonfw/devon4j/blob/master/documentation/guide-beanmapping.asciidoc#bean-mapping[guide] in the devon4j documentation. At the point, the guide is recommending dozer or orika framework the Quarkus usage of bean mapper diverge from devon4j. The frameworks that will be recommended for devon4j, can't be used for a quarkus application because they are using Java reflections. They are not supported in GraalVm right now and causes problems building native applications. That's why we recommend using https://mapstruct.org/[MapStruct] with Quarkus. MapStruct is a code generator that greatly simplifies the implementation of mappings between Java bean types based on a convention over configuration approach. The mapping code will be genreated at compile-time and uses plain method invocations and thus is fast, type-safe, and easy to understand. MapStruct has to be configured to not use Java reflections but it will be shown in this guide.
+
+You can find the official
+https://mapstruct.org/documentation/stable/reference/pdf/mapstruct-reference-guide.pdf[MapStruct reference guide] and a general introduction to MapStruct not in a quarkus context from https://www.baeldung.com/mapstruct[Baeldung].
+
+== MapStruct Dependency 
+To get access to MapStruct we have to add the dependency to our POM.xml:
+
+[source, xml]
+----
+<dependency>
+  <groupId>org.mapstruct</groupId>
+  <artifactId>mapstruct</artifactId>
+  <version>1.4.2.Final</version>
+  <scope>provided</scope>
+</dependency>
+----
+
+MapStruct provides an annotation processor that also has to be added to the POM.xml 
+[source, xml]
+----
+<plugin>
+    <groupId>org.apache.maven.plugins</groupId>
+    <artifactId>maven-compiler-plugin</artifactId>
+    <version>3.8.1</version>
+    <configuration>
+        <source>1.8</source>
+        <target>1.8</target>
+        <annotationProcessorPaths>
+            <path>
+                <groupId>org.mapstruct</groupId>
+                <artifactId>mapstruct-processor</artifactId>
+                <version>1.4.2.Final</version>
+            </path>
+          </path>
+      </annotationProcessorPaths>
+    </configuration>
+</plugin>
+----
+
+=== Lombok 
+MapStruct takes advantage of generated getters, setters, and constructors from Lombok and uses them to
```

Comment:
But MapStruct also works without using Lombok in the project. So make clear that this part is only needen if Lombok is used in the project

Reply:
I also would rather create a separate guide for Lombok and here we can only link and say. "MapStruct also works with link:guide-lombok.asciidoc#mapstruct[Lombok]."

Reply:
I will exclude the Lombok part and put it in a new Guide.